File: app/src/main/java/com/mxt/anitrend/adapter/pager/index/SearchPageAdapter.java
package com.mxt.anitrend.adapter.pager.index;

import android.content.Context;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.mxt.anitrend.R;
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter;
import com.mxt.anitrend.util.KeyUtil;
import com.mxt.anitrend.util.Settings;
import com.mxt.anitrend.view.fragment.search.CharacterSearchFragment;
import com.mxt.anitrend.view.fragment.search.MediaSearchFragment;
import com.mxt.anitrend.view.fragment.search.StaffSearchFragment;
import com.mxt.anitrend.view.fragment.search.StudioSearchFragment;
import com.mxt.anitrend.view.fragment.search.UserSearchFragment;

/**
 * Created by max on 2017/12/19.
 */

public class SearchPageAdapter extends BaseStatePageAdapter {

    public SearchPageAdapter(FragmentManager fragmentManager, Context context) {
        super(fragmentManager, context);
        setPagerTitles(new Settings(context).isAuthenticated()? R.array.search_titles_auth : R.array.search_titles);
    }

    /**
     * Return the Fragment associated with a specified position.
     *
     * @param position
     */
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return MediaSearchFragment.newInstance(getParams(), KeyUtil.ANIME);
            case 1:
                return MediaSearchFragment.newInstance(getParams(), KeyUtil.MANGA);
            case 2:
                return StudioSearchFragment.newInstance(getParams());
            case 3:
                return StaffSearchFragment.newInstance(getParams());
            case 4:
                return CharacterSearchFragment.newInstance(getParams());
            case 5:
                return UserSearchFragment.newInstance(getParams());
        }
        return null;
    }
}

File: app/src/main/java/com/mxt/anitrend/adapter/pager/index/SearchPageAdapter.kt
package com.mxt.anitrend.adapter.pager.index

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.mxt.anitrend.R
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter
import com.mxt.anitrend.extension.LAZY_MODE_UNSAFE
import com.mxt.anitrend.extension.koinOf
import com.mxt.anitrend.ui.fragmentByTagOrNew
import com.mxt.anitrend.ui.model.FragmentItem
import com.mxt.anitrend.util.KeyUtil
import com.mxt.anitrend.util.Settings
import com.mxt.anitrend.view.fragment.search.CharacterSearchFragment
import com.mxt.anitrend.view.fragment.search.MediaSearchFragment
import com.mxt.anitrend.view.fragment.search.StaffSearchFragment
import com.mxt.anitrend.view.fragment.search.StudioSearchFragment
import com.mxt.anitrend.view.fragment.search.UserSearchFragment

/**
 * Created by max on 2017/12/19.
 */
class SearchPageAdapter(
    fragmentActivity: FragmentActivity,
    context: Context,
) : BaseStatePageAdapter(fragmentActivity, context) {
    init {
        val settings = koinOf<Settings>()
        setPagerTitles(
            if (settings.isAuthenticated) {
                R.array.search_titles_auth
            } else {
                R.array.search_titles
            },
        )
    }

    private val fragmentItems by lazy(LAZY_MODE_UNSAFE) {
        listOf<FragmentItem<Fragment>>(
            FragmentItem(
                MediaSearchFragment::class.java,
                Bundle(params).apply {
                    putString(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                },
                "MediaSearchFragmentAnime",
            ),
            FragmentItem(
                MediaSearchFragment::class.java,
                Bundle(params).apply {
                    putString(KeyUtil.arg_mediaType, KeyUtil.MANGA)
                },
                "MediaSearchFragmentManga",
            ),
            FragmentItem(StudioSearchFragment::class.java, Bundle(params)),
            FragmentItem(StaffSearchFragment::class.java, Bundle(params)),
            FragmentItem(CharacterSearchFragment::class.java, Bundle(params)),
            FragmentItem(UserSearchFragment::class.java, Bundle(params)),
        )
    }

    override fun createFragment(position: Int): Fragment = fragmentItems[position].fragmentByTagOrNew(fragmentActivity)
}
